feat: add initial support for stripe payment intents

pshiu opened the pull request:
## Description

This is a draft PR designed to solicit comments on our work to add support for Stripe Payments (or Payment Intents) to the Payment MFE.

This work was required due to a pressing business need to migrate to Stripe in a short time period.

## Additional information

* Jira: https://2u-internal.atlassian.net/browse/REV-3024
* See also: https://github.com/openedx/ecommerce/pull/3847

julianajlk committed:
feat: Install Stripe package and add project-zebra branch to run CI (#637)

johnnagro committed:
build: testing sandbox changes

julianajlk committed:
feat: Initial use of enableStripePaymentProcessor flag (#638)

REV-3034

wdrussell2015 committed:
feat: Add Stripe Elements to Checkout Page(disabled Cybersource for now) (#639)

Project Zebra

julianajlk committed:
feat: Handle Stripe full form data on submit (#640)

christopappas committed:
temp: put test creds in prod .env for sandbox

christopappas committed:
temp: put test creds in prod .env for sandbox

julianajlk committed:
refactor: Use redux-form for billing address to better handle form data (#641)

julianajlk committed:
refactor: Payment form for Stripe and CyberSource (#644)

REV-3001

wdrussell2015 committed:
refactor: Separate stripe client secret from cybersource capture context (#645)

REV-3041

julianajlk committed:
feat: Add Stripe Custom Actions Beta, handle Enterprise bulk enrollments data, and billing form skeleton (#647)

REV-3077

pshiu committed:
fix: follow redirect sent by ecommerce on successful stripe payment (#650)

This is a bit of a hack because the receipt page is hit twice. Tried
using Axios' maxRedirects = 0, but that still did not catch the
redirect.

julianajlk committed:
fix: Remove hard coded Stripe key and update env files (#652)

grmartin commented on README.rst:
```diff
@@ -215,3 +215,9 @@ If you would like to run this frontend against stage.edx.org you can run ``npm r
    :target: https://github.com/openedx/frontend-app-payment/actions/workflows/ci.yml
 .. |license| image:: https://img.shields.io/npm/l/@edx/frontend-app-payment.svg
    :target: @edx/frontend-app-payment
+
```
still needed?

wittjeff commented on src/payment/checkout/Checkout.jsx:
```diff
@@ -64,15 +72,69 @@ class Checkout extends React.Component {
 
+  renderBillingFormSkeleton() {
+    return (
+      <>
```
Is this a shell for tabular data?  If so it might need some ARIA roles.

julianajlk replied:
@wittjeff that's a React Fragment, it will not be rendered in the DOM.

wittjeff commented on src/payment/checkout/payment-form/PaymentForm.jsx:
```diff
@@ -217,40 +214,12 @@ export class PaymentFormComponent extends React.Component {
-              }
-            </div>
-          </div>
+          <PlaceOrderButton
```
It looks like you're replacing a Paragon stateful button with one provided by Stripe. Is that right? I note this because conveying state info to screen reader users is a bit tricky.  No specific requests but something I'll want to test more.

julianajlk replied:
@wittjeff apologies for the delay in this response! We just refactored the existing button to its own component, so it can be used by different parent components. The new component still uses Paragon's `StatefulButton`

wdrussell2015 committed:
fix: false alerts and errors for stripe removed (#655)

REV-3057

pshiu committed:
fix: receive 200s or 400s instead of 302s from ecommerce /checkout (#657)

We tried having Ecommerce send 302s to redirect the Payment MFE to the
receipt page. This didn't work because the library that we use for
requests on the frontend, Axios, doesn't appear to support not following
redirects.

This created a host of CORS and authentication issues due to logging
into Ecommerce from the Payment MFE domain.

We've changed ecommerce's /payment/stripe/checkout to send a HTTP 200 or
400 instead of a 302, along with the information needed for the
redirect. This commit instructs the frontend on what to do when
receiving these 200 or 400s.

wdrussell2015 committed:
fix: dont display confirm payment button until stripe has loaded (#660)

REV-3107

julianajlk committed:
fix: Add trailing slash to STRIPE_RESPONSE_URL (#656)

julianajlk committed:
fix: Hide country zip code from Stripe form to avoid duplicate entry (#658)

REV-3064

pshiu committed:
Revert "temp!: temporarily required ascii characters in name fields" (#553)

This reverts commit 580e1cef26551efa14c6edf6feb378b0c2914d66.

pshiu committed:
fix: add ISSUE_ERROR Redux action to call <FallbackErrorMessage> (#663)

The correct thing to do would be to move all the state logic in
<StripePaymentForm> into Redux. Since we are in a lethal rush, we are
instead implementing a hack, which is bringing Redux to
<StripePaymentForm>.

julianajlk committed:
fix: Change zip code field to be required for Stripe for certain countries (#664)

REV-3064

pshiu committed:
fix: show <TransactionDeclined> on Stripe & backend errors (#665)

christopappas committed:
use locale to display stripe forms in another language (#666)

* feat: use locale to update stripe form language

* docs: leave a comment explaining why code was moved

julianajlk committed:
Revert "Revert "temp!: temporarily required ascii characters in name fields" (#553)" (#668)

This reverts commit c11933ca0e7224802867d6e673482db8ab0d10e9.

wdrussell2015 committed:
refactor: Updated relevant track events for stripe (#669)

REV-3128

pshiu commented:
Thank you very much for your comments! We will be adding these as tickets to our backlog. Some of our teams need this work deployed but sleeping under a flag in production, so we will merge this PR earlier than scheduled to production.

wdrussell2015 reviewed (approved):
Approving for Rollout
